### PR TITLE
Added ClangAnalyzer

### PR DESCRIPTION
**Pull request by andreydanin**
Added support for clang-analyzer plists (tested with plist-multi-file output
format).

### CODE REVIEW

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/io/jenkins/plugins/analysis/warnings/ClangAnalyzer.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=4831466&bid=16051373)

File: pom.xml [hunk 1]
```diff
@@ -33,8 +33,8 @@
     <module.name>${project.groupId}.warnings.ng</module.name>
 
     <codingstyle.version>0.3.2</codingstyle.version>
-    <analysis-model-api.version>8.0.0-beta2</analysis-model-api.version>
-    <analysis-model.version>8.0.0-beta2</analysis-model.version>
+    <analysis-model-api.version>8.0.0-beta3-SNAPSHOT</analysis-model-api.version>
+    <analysis-model.version>8.0.0-beta3-SNAPSHOT</analysis-model.version>
```
Comment (uhafner) [inline pom.xml]: I published analysis-model-api-8.0.0-beta4, please retry.

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/io/jenkins/plugins/analysis/warnings/ClangAnalyzer.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=4831466&bid=16051373)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/io/jenkins/plugins/analysis/warnings/ClangAnalyzer.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=4831466&bid=16051373)

File: src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java [hunk 2]
```diff
@@ -418,6 +418,12 @@ public void shouldProvideMessagesAndDescriptionForSecurityIssuesWithSpotBugs() {
                 "java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input");
     }
 
+    /** Runs the Clang-Analyzer parser on an output file that contains 3 issues. */
+    @Test
+    public void shouldFindAllClangAnalyzerIssues() {
+        shouldFindIssuesOfTool(6, new ClangAnalyzer(), "ClangAnalyzer.txt");
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java]: I think you have a typo in the filename: 
- the file is: `ClangAnalzer.txt`
- the fileName is: `ClangAnalyzer.txt`

Comment (andreydanin) [reply src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java]: Thanks. I've also found the problem in log. Wonder why it wasn't triggered on my machine.

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/io/jenkins/plugins/analysis/warnings/ClangAnalyzer.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=4831466&bid=16051373)

File: src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java (hunk 2 above)
Comment (andreydanin) [reply src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java]: And also there is 3 issues in the test file...

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/main/java/io/jenkins/plugins/analysis/warnings/ClangAnalyzer.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=4831466&bid=16051373)

File: src/main/java/io/jenkins/plugins/analysis/warnings/ClangAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        public String getDisplayName() {
+            return Messages.Warnings_ClangAnalyzer_ParserName();
+        }
+
+        @Override
+        public String getHelp() {
+            return "Use options --analyze --analyzer-output plist-multi-file";
+        }
+
+       @Override
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/ClangAnalyzer.java]: Codacy Issue found: ['method def modifier' has incorrect indentation level 7, expected level should be 8.](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=4831466)

**Comment by uhafner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/io/jenkins/plugins/analysis/warnings/ClangAnalyzer.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=4831466&bid=16051373)

**Comment by uhafner** [comment]
Thank you very much!

**Pull request merged**